server/src/main/java/org/opensearch/search/aggregations/metrics/SumAggregator.java

```diff
@@ -136,6 +140,35 @@ public void collect(int doc, long bucket) throws IOException {
     public LeafBucketCollector getStarTreeCollector(LeafReaderContext ctx, LeafBucketCollector sub, CompositeIndexFieldInfo starTree)
         throws IOException {
         final CompensatedSum kahanSummation = new CompensatedSum(sums.get(0), 0);
+        if (parent != null && subAggregators.length == 0) {
```

Comment:
Any reason why this can't be the default collector regardless of parent collectors ? 

when there is only metric aggregation, there will be a single bucket otherwise multiple buckets.